### PR TITLE
Add mysql & httpd docs

### PR DESCRIPTION
**Pull request by Mrod1598**

### CODE REVIEW

File: docs/mysql.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+# `mysql` Metrics Receiver
+
+The mysql receiver can retrieve stats from your mysql instance using the `mod_status` endpoint.
```
Comment (djaglowski) [inline docs/mysql.md:3]:
```suggestion
The mysql receiver can retrieve stats from your mysql instance by connecting as a monitoring user.
```

File: docs/mysql.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+
+## Prerequisites
+
+Monitoring user must be granted the following permissions:
```
Comment (djaglowski) [inline docs/mysql.md:8]:
```suggestion
It is recommended that you create a dedicated monitoring user. The monitoring user must be granted the following permissions:
```

File: docs/mysql.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+| `password`            |                           | password used to connect to the server
+
+Example Configuration:
+
+```yaml
+metrics:
+  receivers:
+    mysql_metrics:
+      type: mysql
+      stub_status_url: localhost:3306
```
Comment (djaglowski) [inline docs/mysql.md:37]:
```suggestion
      endpoint: localhost:3306
```

File: docs/mysql.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+## Configuration
+
+Following the guide for [Configuring the Ops Agent](https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#file-location), add the required elements for your mysql configuration.
+
+To configure a receiver for your mysql metrics, specify the following fields:
+
+| Field                 | Default                   | Description |
+| ---                   | ---                       | ---         |
+| `type`                | required                  | Must be `mysql`. |
+| `stub_status_url`     | `localhost:3306`          | The url exposed by mysql |
```
Comment (djaglowski) [inline docs/mysql.md:25]:
```suggestion
| `endpoint`     | `localhost:3306`          | The url exposed by mysql |
```

File: docs/mysql.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+| `collection_interval` | `60s`                     | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
+| `username`            |                           | username used to connect to the server
+| `password`            |                           | password used to connect to the server
```
Comment (djaglowski) [inline docs/mysql.md:28]: For consistency, let's user complete sentences, with appropriate capitalization and punctuation.

**Review by djaglowski — COMMENTED** [review]
I reviewed only the metric parts, and have a few comments.

@dehaansa Will you review the log parts at the appropriate time?

File: docs/mysql.md (hunk 1 above)
Comment (Mrod1598) [reply docs/mysql.md]: updated

File: docs/mysql.md (hunk 2 above)
Comment (Mrod1598) [reply docs/mysql.md]: updated

File: docs/mysql.md (hunk 3 above)
Comment (Mrod1598) [reply docs/mysql.md]: updated

File: docs/mysql.md (hunk 4 above)
Comment (Mrod1598) [reply docs/mysql.md]: updated

File: docs/mysql.md (hunk 5 above)
Comment (Mrod1598) [reply docs/mysql.md]: updated

File: docs/httpd.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+The Ops Agent collects the following metrics from your httpd instances.
+
+| Metric                                            | Data Type | Unit        | Labels              | Description |
+| ---                                               | ---       | ---         | ---                 | ---         | 
+| workload.googleapis.com/httpd.connections         | gauge     | connections |                     | The number of active connections currently attached to the HTTP server.  |
+| workload.googleapis.com/httpd.requests            | sum       | Requests    |                     | Total requests serviced by the HTTP server.  |
+| workload.googleapis.com/httpd.workers             | gauge     | connections | state               | The number of workers currently attached to the HTTP server |
+| workload.googleapis.com/httpd.scoreboard          | gauge     | Scoreboard  | state               | Apache HTTP server scoreboard values. |
+| workload.googleapis.com/httpd.bytes               | gauge     | By/s        |                     | Apache HTTP server scoreboard values. |
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply docs/httpd.md]: missed traffic and uptime, whoops. updated.

File: docs/httpd.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+| `collection_interval` | `60s`                     | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
+
+Example Configuration:
+
+```yaml
+metrics:
+  receivers:
+    httpd_metrics:
+      type: httpd
+      stub_status_url: http://localhost:8080/server-status?auto
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply docs/httpd.md]: updated

File: docs/httpd.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+## Configuration
+
+Following the guide for [Configuring the Ops Agent](https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#file-location), add the required elements for your httpd configuration.
+
+To configure a receiver for your httpd metrics, specify the following fields:
+
+| Field                 | Default                   | Description |
+| ---                   | ---                       | ---         |
+| `type`                | required                  | Must be `httpd`. |
+| `stub_status_url`     | `http://localhost:8080/server-status?auto` | The url exposed by the httpd stats module. |
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply docs/httpd.md]: updated

File: docs/httpd.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
+# `apache httpd` Metrics Receiver
+
+The httpd receiver can retrieve stats from your httpd instance using the `/server-status?auto` endpoint.
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply docs/httpd.md]: updated

File: docs/httpd.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+Access logs contain the [`httpRequest` field](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#httprequest):
+
+<!-- TODO: Validate that we can get all of these for apache -->
+
+| Field | Type | Description |
+| ---   | ---- | ----------- |
+| `httpRequest.protocol` | string | Protocol used for the request |
+| `httpRequest.referer` | string | Contents of the `Referer` header |
+| `httpRequest.remoteIp` | string | Client IP address |
```
Comment (dehaansa) [inline docs/httpd.md:108]: No remoteIP in apache default log config.

File: docs/httpd.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+| `httpRequest.requestMethod` | string | HTTP method |
+| `httpRequest.requestUrl` | string | Request URL (typically just the path part of the URL) |
+| `httpRequest.responseSize` | string (`int64`) | Response size |
+| `httpRequest.status` | number | HTTP status code |
+| `httpRequest.userAgent` | string | Contents of the `User-Agent` header |
+| `jsonPayload.host` | string | Contents of the `Host` header (usually not reported by httpd) |
```
Comment (dehaansa) [inline docs/httpd.md:114]: remove the "usually not reported by httpd"

File: docs/httpd.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+Error logs contain the following fields in the [`LogEntry`](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry):
+
+<!-- TODO: Validate that we can get all of these for apache -->
+
+| Field | Type | Description |
+| ---   | ---- | ----------- |
+| `jsonPayload.level` | string | Log entry level |
+| `jsonPayload.pid` | number | Process ID |
+| `jsonPayload.tid` | number | Thread ID |
+| `jsonPayload.connection` | number | Connection ID |
```
Comment (dehaansa) [inline docs/httpd.md:129]: No connection field for apache.

File: docs/httpd.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+| `jsonPayload.message` | string | Log message |
+| `jsonPayload.client` | string | Client IP address (optional) |
+| `jsonPayload.server` | string | httpd server name (optional) |
```
Comment (dehaansa) [inline docs/httpd.md:132]: No server field for apache

Comment (dehaansa) [reply docs/httpd.md:132]: or request, subrequest, upstream, host, referrer

File: docs/httpd.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+| `httpRequest.userAgent` | string | Contents of the `User-Agent` header |
+| `jsonPayload.host` | string | Contents of the `Host` header (usually not reported by httpd) |
+| `jsonPayload.user` | string | Authenticated username for the request |
+| `timestamp` | string ([`Timestamp`](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp)) | Time the request was received |
+
+Any fields that are blank or missing will not be present in the log entry.
+
+Error logs contain the following fields in the [`LogEntry`](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry):
+
+<!-- TODO: Validate that we can get all of these for apache -->
```
Comment (dehaansa) [inline docs/httpd.md:122]: Add 'jsonPayload.module', string, apache module where the log originated
Add 'jsonPayload.errorCode', string, apache error code

Comment (Mrod1598) [reply docs/httpd.md:121]: added, added

File: docs/httpd.md (hunk 13 above)
Comment (Mrod1598) [reply docs/httpd.md]: removed all

File: docs/httpd.md (hunk 12 above)
Comment (Mrod1598) [reply docs/httpd.md]: removed

File: docs/httpd.md (hunk 11 above)
Comment (Mrod1598) [reply docs/httpd.md]: removed

File: docs/httpd.md (hunk 10 above)
Comment (Mrod1598) [reply docs/httpd.md]: removed

**Pull request closed without merging**